cmd/prometheus/main.go

```diff
@@ -118,7 +122,7 @@ func Main() int {
 		GoVersion: version.GoVersion,
 	}
 
-	webHandler := web.New(memStorage, queryEngine, targetManager, ruleManager, version, flags, &cfg.web)
+	webHandler := web.New(memStorage, queryEngine, queryCtx, targetManager, ruleManager, version, flags, &cfg.web)
```

Comment:
The web API already has an `Options` struct, but it's really more simple options (not complex instantiated types) coming from flags so far:

```go
// Options for the web Handler.
type Options struct {
	ListenAddress        string
	ExternalURL          *url.URL
	RoutePrefix          string
	MetricsPath          string
	UseLocalAssets       bool
	UserAssetsPath       string
	ConsoleTemplatesPath string
	ConsoleLibrariesPath string
	EnableQuit           bool
}
```

It's instantiated as part of the main `cfg` struct. Still maybe better to reuse this struct and `=` assign the missing values in `main()` rather than having a second options struct for the other parameters?